substrate/frame/broker/src/dispatchable_impls.rs

```diff
@@ -91,6 +91,15 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
+	pub(crate) fn do_remove_lease(task: TaskId) -> DispatchResult {
+		let mut r = Leases::<T>::get();
+		let i = r.iter().position(|lease| lease.task == task).ok_or(Error::<T>::LeaseNotFound)?;
```

Comment:
`retain` could make for cleaner code here

Reply:
`retain` would iterate the entire vector instead of ending directly when the `task` was found.